File: 0624-maximum-distance-in-arrays/0624-maximum-distance-in-arrays.cpp
class Solution {
public:
    int maxDistance(vector<vector<int>>& arrays) {  
        int max_ = 0;  
        int curr_min = arrays[0][0];
        int curr_max = arrays[0].back(); 
        for(int i = 1; i < arrays.size(); i++){
            int first = abs(curr_max - arrays[i][0]); 
            int second = abs(curr_min - arrays[i].back());
            max_ = max(max_, max(first, second)); 
            curr_min = min(curr_min, arrays[i][0]); 
            curr_max = max(curr_max, arrays[i].back());          
        }

        return max_; 
    }
};

File: 0624-maximum-distance-in-arrays/0624-maximum-distance-in-arrays.java
class Solution {
    public int maxDistance(List<List<Integer>> arrays) {
        int smallest = arrays.get(0).get(0); 
        int largest = arrays.get(0).getLast(); 
        int answer = 0; 
        for(int i = 1; i < arrays.size(); i++){
            answer = Math.max(answer, Math.abs(arrays.get(i).getLast() - smallest)); 
            answer = Math.max(answer, Math.abs(arrays.get(i).get(0) - largest)); 
            smallest = Math.min(smallest, arrays.get(i).get(0)); 
            largest = Math.max(largest, arrays.get(i).getLast()); 
        }
        return answer; 
    }
}